src/main/java/net/jodah/typetools/TypeResolver.java

```diff
@@ -47,15 +51,43 @@
   private static volatile boolean CACHE_ENABLED = true;
   private static boolean RESOLVES_LAMBDAS;
   private static Method GET_CONSTANT_POOL;
+  private static Method GET_CONSTANT_POOL_SIZE;
+  private static Method GET_CONSTANT_POOL_METHOD_AT;
   private static final Map<String, Method> OBJECT_METHODS = new HashMap<String, Method>();
   private static final Map<Class<?>, Class<?>> PRIMITIVE_WRAPPERS;
   private static final Double JAVA_VERSION;
 
   static {
     JAVA_VERSION = Double.parseDouble(System.getProperty("java.specification.version", "0"));
+
     try {
+      Unsafe unsafe = AccessController.doPrivileged(new PrivilegedExceptionAction<Unsafe>() {
+        @Override
+        public Unsafe run() throws Exception {
+          final Field f = Unsafe.class.getDeclaredField("theUnsafe");
+          f.setAccessible(true);
+
+          return (Unsafe) f.get(null);
+        }
+      });
+
       GET_CONSTANT_POOL = Class.class.getDeclaredMethod("getConstantPool");
-      GET_CONSTANT_POOL.setAccessible(true);
+      String constantPoolName = JAVA_VERSION < 9 ? "sun.reflect.ConstantPool" : "jdk.internal.reflect.ConstantPool";
+      Class<?> constantPoolClass = Class.forName(constantPoolName);
+      GET_CONSTANT_POOL_SIZE = constantPoolClass.getDeclaredMethod("getSize");
+      GET_CONSTANT_POOL_METHOD_AT = constantPoolClass.getDeclaredMethod("getMethodAt", int.class);
+
+      // setting the methods as accessible
+      Field overrideField = AccessibleObject.class.getDeclaredField("override");
```

Comment:
It's a shame that we need to depend on a private field name here. Can you explain why this approach is necessary versus the traditional `Method.setAccessible(true)` approach?